frontend/src/common/hooks/useCategoryFilter/useCategoryFilter.ts

```diff
@@ -0,0 +1,897 @@
+import { useCallback, useEffect, useState } from "react";
+import { Filters, Row } from "react-table";
+import { Ontology } from "src/common/entities";
+import {
+  buildCuratedOntologyCategoryView,
+  onFilterCuratedOntologyCategory,
+} from "src/common/hooks/useCategoryFilter/common/curatedOntologyUtils";
+import {
+  buildMultiPanelCategoryView,
+  buildMultiPanelCurrentFilters,
+  buildMultiPanelSelectedUIState,
+  buildMultiPanelUIState,
+  isMultiPanelCategoryFilterConfig,
+  onFilterMultiPanelCategory,
+} from "src/common/hooks/useCategoryFilter/common/multiPanelOntologyUtils";
+import {
+  getCategoryFilter,
+  isCategoryValueIdSet,
+} from "src/common/hooks/useCategoryFilter/common/utils";
+import {
+  CATEGORY_FILTER_CONFIGS_BY_ID,
+  COLLATOR_CASE_INSENSITIVE,
+} from "src/components/common/Filter/common/constants";
+import {
+  Categories,
+  CategoryFilter,
+  CategoryFilterConfig,
+  CategoryFilterPanelConfig,
+  CategorySet,
+  CategoryValueId,
+  CategoryView,
+  CATEGORY_FILTER_ID,
+  CuratedOntologyCategoryFilterConfig,
+  FilterKey,
+  FilterState,
+  KeyedSelectCategoryValue,
+  MultiPanelCategoryFilterUIState,
+  MultiPanelSelectedUIState,
+  MultiPanelUIState,
+  OnFilterFn,
+  OntologyNode,
+  OntologyTermSet,
+  ONTOLOGY_VIEW_KEY,
+  ON_FILTER_SOURCE,
+  Range,
+  RangeCategory,
+  SelectCategoryValue,
+} from "src/components/common/Filter/common/entities";
+import { listOntologyTreeIds } from "src/components/common/Filter/common/utils";
+import {
+  addRangeCategories,
+  buildRangeCategoryView,
+  isSelectedValueRange,
+  onFilterRangeCategory,
+} from "./common/rangeUtils";
+import {
+  buildSelectCategoryView,
+  onFilterSelectCategory,
+} from "./common/selectUtils";
+
+/**
+ * Shape of return value from this useFilter hook.
+ */
+export interface FilterInstance {
+  categoryViews: CategoryView[];
+  multiPanelSelectedUIState: MultiPanelSelectedUIState;
```

Comment:
Added an accessor to enable hook callee's to save the selected state of multi-panel category filters to local storage.